### PR TITLE
docs(using_deis): explain difference between web and cmd

### LINKED ISSUE #2861: surprising effect of changing from go-buildpack to Dockerfile
I was using the go buildpack and I added a Dockerfile to switch to using that style of build.  I did this in an existing project.

The surprising result is that my containers built fine but the container would not start,  No output from the logs at all.  I eventually found the problem when `deis run <start command>`  succeeded in starting the container with my program inside it but the "automatic" deployment did not.

It wasn't being started because the "dyno type" had gotten set to web because of the original go build pack and the Dockerfile expected a type of "cmd."  `deis ps:scale cmd=1` fixes the problem, but even a warning message might be helpful.

### PR DESCRIPTION
**Pull request by bacongobbler**
Also states that Deis does not automatically migrate from web to
cmd process types when switching contexts.

closes #2861

### CODE REVIEW

File: docs/using_deis/process-types.rst [hunk 1]
```diff
@@ -92,4 +92,20 @@ Use ``deis scale web=3`` to increase ``web`` processes to 3, for example. Scalin
...
 running that process.
 
+Web vs Cmd Process Types
+------------------------
+
+When deploying to Deis using a Heroku Buildpack, Deis boots the ``web`` process type to boot the
+application server. When you deploy an application that has a Dockerfile or uses :ref:`Docker
+images <using-docker-images>`, Deis boots the ``cmd`` process type. Both are identical to
+each other in that they are exposed to the router as web applications. However, The ``cmd`` process
+type is special in which it is equivalent to running the :ref:`container` without any additional
```
Comment (mboersma) [inline docs/using_deis/process-types.rst]: I would replace "in which" by "in that" or "because".

File: docs/using_deis/process-types.rst [hunk 2]
```diff
@@ -92,4 +92,20 @@ Use ``deis scale web=3`` to increase ``web`` processes to 3, for example. Scalin
 process type directly changes the number of :ref:`Containers <container>`
 running that process.
 
+Web vs Cmd Process Types
+------------------------
+
+When deploying to Deis using a Heroku Buildpack, Deis boots the ``web`` process type to boot the
+application server. When you deploy an application that has a Dockerfile or uses :ref:`Docker
+images <using-docker-images>`, Deis boots the ``cmd`` process type. Both are identical to
```
Comment (mboersma) [inline docs/using_deis/process-types.rst]: Maybe instead of "Both are identical to each other" say "Both act similarly"?

File: docs/using_deis/process-types.rst [hunk 3]
```diff
@@ -92,4 +92,20 @@ Use ``deis scale web=3`` to increase ``web`` processes to 3, for example. Scalin
...
+each other in that they are exposed to the router as web applications. However, The ``cmd`` process
+type is special in which it is equivalent to running the :ref:`container` without any additional
+arguments. Every other process type is equivalent to running the relevant command that is provided
+in the Procfile.
+
+When migrating from Heroku Buildpacks to a Docker-based deployment, Deis does not explicitly
```
Comment (mboersma) [inline docs/using_deis/process-types.rst]: Maybe rephrase to ",Deis will not convert ``web`` process types to ``cmd``. To do this,"

This is all good, just trying to reduce verbiage. :pencil2:

Comment (bacongobbler) [inline docs/using_deis/process-types.rst]: All good input. I suck at articulating my thoughts, so this is all very appreciated

**Comment by mboersma** [comment]
Code LGTM.

**Comment by carmstrong** [comment]
Much cleaner with @mboersma's comments. Good work, gents.

:ship:
